test/HtmlParser.test.js

```diff
@@ -252,4 +252,71 @@ describe("HtmlParser", function(){
       expect(calledData).to.equal(2)
     })
   })
+  describe("preserve non-breaking space option", () => {
+    it("should preserve spaces when in text", function(){
+      let calledData = "";
+      const parser = new HtmlParser({preserveNbsp: true});
+      parser.on("data", data => {
+        if (data.text) calledData += data.text;
+      });
+      parser.end("Title: <b> Jan  Bananberg</b>");
+      expect(calledData).to.equal("Title:  Jan  Bananberg");
+    })
+    it("should preserve spaces when in text across chunks", function(){
+      let calledData = "";
+      const parser = new HtmlParser({preserveNbsp: true});
+      parser.on("data", data => {
+        if (data.text) calledData += data.text;
+      });
+      parser.write("Title: ");
+      parser.write("<b> Jan");
+      parser.write("  Bananb");
+      parser.end("erg</b>");
+      expect(calledData).to.equal("Title:  Jan  Bananberg");
+    })
+    it("should still ignore other whitespace chars when in text", function(){
+      let calledData = "";
+      const parser = new HtmlParser({preserveNbsp: true});
+      parser.on("data", data => {
+        if (data.text) calledData += data.text;
+      });
+      parser.end("Title:\r\n<b> Jan \r\n\t Bananberg</b>");
+      expect(calledData).to.equal("Title: Jan  Bananberg");
+    })
```

Comment:
Preserving space will be a bit more complicated since the default is to collapse whitespace to a single space, which runs into problems in the above case. I think the best option is to perhaps support turning collapse whitespace off. Remember that this is a stream, so you can always transform your nodes object-by-object outside of this library:

```
// do post-processing on each node in myTransform
const parser = new HtmlParser({preserveWS: true})
myFile.pipe(parser).pipe(myTransform)
```

Where your transform reads the object stream returned by atlas-html-stream. I think this is the best option to keep the html parser performant. I think if we turn off collapsing all WS, we won't need to support the `pre` tag because we will return the content as-is anyway.

Reply:
I agree!